File: python3/1.two-sum.py
#
# @lc app=leetcode id=1 lang=python3
#
# [1] Two Sum
#

# @lc code=start
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dict = {}
        for i in range(len(nums)):
            if target - nums[i] in dict:
                return [dict[target-nums[i]], i]
            dict[nums[i]] = i
        return [-1,-1]
# @lc code=end

File: cpp/1.two-sum.cpp
/*
 * @lc app=leetcode id=1 lang=cpp
 *
 * [1] Two Sum
 */

// @lc code=start
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> m;
        for(int i=0; i<nums.size(); ++i){
            if(m.count(target-nums[i]))
                return {m[target-nums[i]], i};
            m[nums[i]] = i;
        }
        return {-1,-1};
    }
};
// @lc code=end

